village/src/App.js

```diff
@@ -14,11 +17,84 @@ class App extends Component {
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
+ //GET
+  componentDidMount() {
+
+
+    axios
+
+      .get("http://localhost:3333/smurfs")
+
+      .then(response => {
+       console.log(response);
+       this.setState({ smurfs: response.data });
+     })
+
+      .catch(err => {
+       console.log(err);
+     });
+ }
+
+  addSmurfListOnServer = smurfData => {
+    this.setState({smurfs: smurfData});
+}
+
+//DELETE
+  deleteSmurfFromServer = id => {
+    axios
+
+    .delete(`http://localhost:3333/smurfs/${id}`)
+
+    .then(response => {
+     console.log(response);
+     this.setState({ smurfs: response.data });
+   })
+
+    .catch(err => {
+     console.log(err);
+   });  
+  }
+  
+//PUT
+
+postSmurfFromServer = (smurf, id) => {
```

Comment:
Your other function names are very semantic (carry clear meaning), but this one is hard to understand. How could you improve this.